dist/dialog.linker.build.js

```diff
@@ -102,13 +113,24 @@ window.jQuery = window.jQuery || window.shoestring;
 			return;
 		}
 
-		// if someone is calling close directly and the hash for this dialog is in
-		// the url then we need to go back, this will trigger the hashchange binding
-		// in init
-		// NOTE the bindings seem better in the constructor e.g.
-		// "#foo".indexOf("foo") === 1
-		if(window.location.hash.replace(/^#/, "") === this.hash){
-			window.history.back();
+		// if close() is called directly and the hash for this dialog is at the end of the url,
+		// then we need to change the hash to remove it, either by going back if we can, or by adding a history
+		// state that doesn't have it at the end
+		if( window.location.hash.split( "#" ).pop() === this.hash ){
+			// let's check if the first segment in the hash is the same as the first segment in the initial hash
+			// if not, it's safe to use back() to close this out and clean the hash up
+			var firstHashSegment = window.location.hash.split( "#" )[ 1 ];
+			var firstInitialHashSegment = this.initialLocationHash.split( "#" )[ 1 ];
+			if( firstHashSegment && firstInitialHashSegment && firstInitialHashSegment !== firstHashSegment ){
+				window.history.back();
+			}
```

Comment:
I think if we end up changing the hash to something other than the initial hash value and then change it back to the initial hash value we would still want to go back. Example:

1. initial hash: `#foo-dialog`
2. close foo dialog: empty hash
3. open foo dialog: `#foo-dialog`
4. should use `window.history.back`